torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -436,8 +436,15 @@ def get_cache_blocking(register_blocking, thread_blocking):
             def get_num_byte(dtype):
                 return torch.tensor([], dtype=dtype).element_size()
 
-            num_byte_A = get_num_byte(self.input_nodes[0].get_dtype())
-            num_byte_B = get_num_byte(self.input_nodes[1].get_dtype())
+            dtype_A = self.input_nodes[0].get_dtype()
+            dtype_B = self.input_nodes[1].get_dtype()
+            num_byte_A = get_num_byte(dtype_A)
+            num_byte_B = get_num_byte(dtype_B)
+            if dtype_A is torch.bfloat16 and dtype_B is torch.int8 and Kr != 1:
+                # We will cache dequantized weights (BF16) in L1D for AMX micro-kernel.
+                # In this case, the choice of the micro-kernel being used can't be decoupled from
+                # the cache blocking
+                num_byte_B *= 2
```

Comment:
Thanks again for reviewing, @jgong5!
Sorry, I couldn't make changes earlier due to dev-machines' downtime.

I changed `num_byte_B *= num_byte_A` but didn't remove `Kr != 1` condition because the AVX512 kernel doesn't currently cache dequantized B. I intend to add another AVX512 micro-kernel that's more efficient for GEMV and small values of `M`, so I could make this change when I'd do so.

However, if you'd insist, I could add these changes now, and also modify the AVX512 micro-kernel.